src/rust/ide/view/graph-editor/src/component/visualization/instance.rs

```diff
@@ -21,8 +21,10 @@ use ensogl::display::Scene;
 #[derive(Clone,CloneRef,Debug)]
 #[allow(missing_docs)]
 pub struct FrpInputs {
-    pub set_size  : frp::Source<Vector2>,
-    pub send_data : frp::Source<Data>,
+    pub set_size   : frp::Source<Vector2>,
+    pub send_data  : frp::Source<Data>,
+    pub activate   : frp::Source,
+    pub deactivate : frp::Source,
 }
```

Comment:
Should this be refactored to use the FRP macro?

Reply:
I'm not sure. The macro makes the `source` field crate-local. But we want to allow anyone to extend the instance network as they please, when implementing "rust" visualizations.

Reply:
Michael is right, this should use FRP macro. Adam is right as well :D The macro should allow thesethings to be public (I wondering if it does not already). Adam, could you add fixme here, please? :)

Reply:
THere is another problem: the Frp structures created by macro owns the Network. But here we create "Visualization API" in existing network (the specific visualization creates and owns its Network).